### DISCUSSION TITLE
run flutter-gallery on imx8 target ( dart-imx8mq dev kit)

### DISCUSSION THREAD

**Discussion by ovjohns** [meta-flutter/meta-flutter#458] [Q&A] [answered]
Hello, 
 I am trying to get the example flutter-gallery working on a dart-imx8mq dev kit. I am using kirkstone yocto and have been able to build the core-weston-image and included flutter-engine and flutter-gallery. I had previously built and tested a hello-world example using a debian image, but our typical workflow uses yocto. 
I can see libflutter_engine.so and libapp.so on my target image. I can also see homescreen in my target image with the current core-image-weston image. 
The issue I am having is that I cannot seem to find the actual flutter-gallery executable. I checked in my yocto work directory and it does not look as though the executable is actually generated. The flutter-gallery directory in work only contains the data and lib directories. 
Question then is : How do I generate an actual executable that can be run on the target. I tried to include packagegroup-flutter-sdk-deps in IMAGE_INSTALL:append, but when I do this, bitbake core-image-weston fails. 
I am not sure what I am doing wrong, or what step I am missing. 
Any input/help would be most appreciated. 
Thanks 
Victor

**Comment by jwinarske** [comment, marked as answer]
@ovjohns try this
<Login as root>
passwd -d weston
exit
<Login as weston>
homescreen --b=/usr/share/flutter/3.19.3/release --f

--b points to the bundle folder
--f sets fullscreen

**Comment by ovjohns** [comment]
Joel,
   Thanks for the quick reply.

There are two issues with the flutter gallery on the imx8 platform that I was having

  1.  There is an issue with xdg-user-dirs not included in the core-weston image. So I included that and re-imaged.
  2.  Once that was done I was seeing another error related to compositing. I had seen a similar error when I was trying to get an initial flutter application running using the Debian distro on the imx8.

The github issue is here : https://github.com/meta-flutter/meta-flutter/issues/275

I applied that patch, rebuild and re-imaged.



With these two changes flutter-gallery now appears to be working on the dart imx8mq dev kit.



Thanks for your help

Best

Victor
